core/client/src/main/java/alluxio/hadoop/AbstractFileSystem.java

```diff
@@ -240,32 +239,28 @@ public long getDefaultBlockSize() {
 
     AlluxioURI path = new AlluxioURI(HadoopUtils.getPathWithoutScheme(file.getPath()));
     List<FileBlockInfo> blocks = getFileBlocks(path);
-
-    List<BlockLocation> blockLocations = new ArrayList<BlockLocation>();
-    for (int k = 0; k < blocks.size(); k++) {
-      FileBlockInfo info = blocks.get(k);
-      long offset = info.getOffset();
-      long end = offset + info.getBlockInfo().getLength();
+    List<BlockLocation> blockLocations = new ArrayList<>();
+    for (FileBlockInfo fileBlockInfo : blocks) {
+      long offset = fileBlockInfo.getOffset();
+      long end = offset + fileBlockInfo.getBlockInfo().getLength();
       // Check if there is any overlapping between [start, start+len] and [offset, end]
       if (end >= start && offset <= start + len) {
-        ArrayList<String> names = new ArrayList<String>();
-        ArrayList<String> hosts = new ArrayList<String>();
-        List<WorkerNetAddress> addrs = Lists.newArrayList();
-        // add the existing in-memory block locations first
-        for (alluxio.wire.BlockLocation location : info.getBlockInfo().getLocations()) {
-          addrs.add(location.getWorkerAddress());
+        ArrayList<String> names = new ArrayList<>();
+        ArrayList<String> hosts = new ArrayList<>();
+        // add the existing in-memory block locations
+        for (alluxio.wire.BlockLocation location : fileBlockInfo.getBlockInfo().getLocations()) {
+          String host = location.getWorkerAddress().getHost();
+          int port = location.getWorkerAddress().getDataPort();
```

Comment:
Oh I mean even in these portions, we could use `HostAndPort` and then avoid hardcoding the conversion to a string (ie. `host + ":" + port`). I think we are ok with trusting HostAndPort's implementation to do the correct thing in `toString`.